decision-records/010-remove-init-method.md

```diff
@@ -0,0 +1,98 @@
+
+This is mitigated by the changes being quite small (removing a `.init()` call following the current instantiation of the component) and only affects a portion of our users, who use our component classes directly rather than through `initAll`.
+
+<!-- Any accepted risks associated with the decision. -->
+
+## Alternatives considered
+
+### Deprecating the init method in v5 for future removal in v6
+
+Because there wasn't really an alternative for users to switch to if we'd deprecated in v4 (if people stopped calling `init()`, there'd be no initialisation), we couldn't make the deprecation in v4.
+
+Implementing an empty `init()` to document its deprecation in v5 would only delay the update users need to make in their code. This will come at the detriment of the next release: on top of focusing on the topic of that new release, people will also have to update their JavaScript (which is the focus of v5).
+
+The scope of the change users have to make, which will be documented, as well as our continued support for v4 during a year after we ship v5 seems enough of a mitigation to directly remove the method.
+
+### Keeping the `init` method to make use of `async`
+
+`init` could be made asynchronous, which would isolate the initialisation of each component and allow to run asynchronous calls inside it.
+
+However:
+
+- Asynchronicity is not something any of our component need. Future components, like an `Autocomplete`, may involve some asynchronicity. However this asynchronicity applies to internals of the components, not their initialisation. For example, even if an Autocomplete component was relying on fetching a list of suggestion from the server, we'd want to update the UI to a text field immediately, synchronously, and delay only the display of these suggestions. Other systems (like events) could be set up to notify service code of the availability of the suggestion, rather than have it `await` the `init` call.
+- Asynchronous JavaScript code also requires a deeper understanding of JavaScript from our users: for example to understand what's running concurrently or in series, to handle errors... The isolation it provides when instantiating a series of components (like in `initAll()`) can already be achieved by catching (and logging) any error that may be thrown, so components after the one erroring still initialise.
+
+<!-- Other options discussed but decided against. -->
+
+## Implications
+
+<!-- Impacts of this decision. -->
+
+### Users need to update their code
+
+Component initialisation will be reduced to `new ComponentClass()` instead of `new ComponentClass().init()`, so users will need to update their code to reflect this.
+
+This will only affect users that were taking responsibility for instantiating components on specific elements themselves. Users who rely on `initAll()` will not have any changes to make as we'll have updated that function's implementation.
```

Comment:
Sorry, felt like the words were hiding the meaning:

```suggestion
The `initAll()` method will not be affected.
```

Reply:
Doesn't hurt to have it there as well if someone gets linked directly to this section (they might not scroll back up).

Reply:
@colinrotherham Updated that bit, which I'd missed 😊

Reply:
Sounds good in isolation, but repeats text from the paragraph above.

How about my new suggestion in https://github.com/alphagov/govuk-design-system-architecture/pull/33#discussion_r1277748683 and https://github.com/alphagov/govuk-design-system-architecture/pull/33#pullrequestreview-1552449851?